src/Simplify.cpp

```diff
@@ -278,17 +286,45 @@ void Simplify::ScopedFact::learn_true(const Expr &fact) {
                 learn_lower_bound(v, i.min);
             }
         }
+    } else if (const Call *c = fact.as<Call>()) {
+        if (c->is_intrinsic(Call::likely) || c->is_intrinsic(Call::likely_if_innermost)) {
+            learn_true(c->args[0]);
+            return;
+        }
     } else if (const And *a = fact.as<And>()) {
         // Both must be true
         learn_true(a->a);
         learn_true(a->b);
+        return;
     } else if (const Not *n = fact.as<Not>()) {
         learn_false(n->a);
-    } else if (simplify->truths.insert(fact).second) {
+        return;
+    }
+    if (simplify->truths.insert(fact).second) {
```

Comment:
Adding compound facts in addition to the leaves looks like it adds quite a lot of redundant facts in the case where you have something like a && b && c && d && e

Reply:
Note I also added an early return in the And case above, so it shouldn't do that in this case. I made this change so we could learn LE/LT/EQ in addition to the bounds/alignment we learn from those.